### DISCUSSION TITLE
Prevent stripping of Refit-endpoint in library

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#16869] [Q&A] [answered]
I have a quite complex setup now. I have one class library containing a Service-class, that calls into a Refit-generated endpoint.
This library gets referenced by a Uno-application.
That Uno-application gets served by a WebAPI-project.

Everything works as expected locally. But after deploying in Release-Mode I get the error "IMyApi doesn't look like a Refit interface". If I set the `WasmShellILLinkerEnabled`-property to false, it works. So there is definitely something being stripped out.

So I tried to add a `LinkerConfig.xml` to the services-library and to the Uno-app. I tried different combinations, tried to stop my IMyApi-type from being trimmed. But none works. Even if I add the whole assembly it doesn't work. My assumption is that I have the wrong type/assembly name.

Does anyone have an idea how exactly I would have to define my linker-config to allow it to work in this setup?

**Comment by francoistanguay** [comment]
Maybe some helpful tips here? https://github.com/reactiveui/refit/issues/1389

**Reply by TopperDEL** [reply, marked as answer]
Thanks, Francosi!

I decided to switch over to [Flurl](https://flurl.dev/), which provides a similar experience in my case. Refit does not seem to support Trimming in the near future. There seems to be not way to overcome the trimming-issues - at least I could not find any.

**Comment by AuderixDev** [comment]
I had the same problems with Refit. Debug mode was all fine, Release mode didn't work... 

I had to add in the LinkerConfig.xml the following lines: 
```
<assembly fullname="Refit" />
<assembly fullname="Service.Sdk" /> <!--- Project where the API Interface is defined -->
```
Because Refit uses Json, i also uncommented/added this lines: 
```
<assembly fullname="System.Net.Http.Json" />
<assembly fullname="System.Text.Json" />
```
Hope this helps
